src/components/form/range/range_levels.styles.ts

```diff
@@ -8,11 +8,49 @@
 
 import { css } from '@emotion/react';
 import { UseEuiTheme } from '../../../services';
+import { EuiRangeLevel } from './range_levels';
 import { euiRangeVariables } from './range.styles';
 
+export const euiRangeLevelColor = (
+  color: EuiRangeLevel['color'],
+  euiTheme: UseEuiTheme['euiTheme']
+) => {
+  switch (color) {
+    case 'primary':
+      return euiTheme.colors.primary;
+    case 'success':
+      return euiTheme.colors.success;
+    case 'warning':
+      return euiTheme.colors.warning;
+    case 'danger':
+      return euiTheme.colors.danger;
+    default:
+      return color;
+  }
```

Comment:
[optional nit] this feels like it could be written more succinctly:

```suggestion
  const isNamedColor = ['primary, 'success', 'warning', 'danger'].includes(color);
  return isNamedColor ? euiTheme.colors[color] : color;
```

Reply:
Wow! 🤯